src/mds/CInode.cc

```diff
@@ -4027,21 +4027,38 @@ void CInode::decode_import(bufferlist::const_iterator& p,
 
 void InodeStoreBase::dump(Formatter *f) const
 {
+  f->open_object_section("inode");
   inode.dump(f);
+  f->close_section();
   f->dump_string("symlink", symlink);
+
+  f->open_array_section("xattrs");
+  for (const auto& [key, val] : xattrs) {
+    f->open_object_section("xattr");
+    f->dump_string("key", key);
+    std::string v(val.c_str(), val.length());
+    f->dump_string("val", v);
+    f->close_section();
+  }
+  f->close_section();
+  f->open_object_section("dirfragtree");
+  dirfragtree.dump(f);
+  f->close_section(); // dirfragtree
+  
   f->open_array_section("old_inodes");
   for (const auto &p : old_inodes) {
     f->open_object_section("old_inode");
     // The key is the last snapid, the first is in the mempool_old_inode
     f->dump_int("last", p.first);
+    f->open_object_section("old_inode_data");
```

Comment:
this section also looks unnecessary

Reply:
@batrick the reason it same to decode inode_t. it's type is old_inode_t.

Reply:
This section too.